client/src/Components/MapComponent.js

```diff
@@ -106,22 +107,50 @@ export default function MapComponent({center, zoom}) {
     }
   }, [center])
 
-  const polylineRef = useRef(null)
-
+  const looppolylineRef = useRef(null)
+  const upperpolylineRef = useRef(null)
   const onMapLoad = ({map, maps}) => {
-    polylineRef.current = new maps.Polyline({
+    looppolylineRef.current = new maps.Polyline({
       path: loopPath,
       geodesic: true,
       strokeColor: '#FF0000',
       strokeOpacity: 1,
       strokeWeight: 4,
     })
+    upperpolylineRef.current = new maps.Polyline({
+      path: upperCampusPath,
+      geodesic: true,
+      strokeColor: '#0000FF',
+      strokeOpacity: 1,
+      strokeWeight: 4,
+    })
+   
+
 
-    polylineRef.current.setMap(map)
+    upperpolylineRef.current.setMap(map)
+    looppolylineRef.current.setMap(map)
   }
 
   useEffect(() => {

```

Comment:
Need a way to make this generic, code sample give below can yive you an example on how to get started

```
// At the start of your component, replace looppolylineRef and upperpolylineRef with polylineRefs
const polylineRefs = useRef({})

const onMapLoad = ({map, maps}) => {
    // This could be fetched dynamically or be an external configuration
    const routes = [{
        name: 'LOOP',
        path: loopPath,
        strokeColor: '#FF0000',
    }, {
        name: 'UPPER CAMPUS',
        path: upperCampusPath,
        strokeColor: '#0000FF',
    }]

    routes.forEach((route) => {
        polylineRefs.current[route.name] = new maps.Polyline({
            path: route.path,
            geodesic: true,
            strokeColor: route.strokeColor,
            strokeOpacity: 1,
            strokeWeight: 4,
        })
        polylineRefs.current[route.name].setMap(map)
    })
}

useEffect(() => {
    // Get all the keys (route names) from polylineRefs
    const routeNames = Object.keys(polylineRefs.current)

    routeNames.forEach(routeName => {
        // For each route, if it is selected, set its opacity to 1, else set it to 0
        if (polylineRefs.current[routeName]) {
            if (selectedRoute.includes(routeName)) {
                polylineRefs.current[routeName].setOptions({strokeOpacity:1})
            } else {
                polylineRefs.current[routeName].setOptions({strokeOpacity:0})
            }
        }
    })
}, [selectedRoute])
```